docs/tooling-consolidation.md

```diff
@@ -0,0 +1,104 @@
+Microsoft.AspNetCore.Razor.Tools
+Microsoft.CodeAnalysis.Razor
+Microsoft.NET.Sdk.Razor
+````
+The following (and associated tests) will remain in aspnetcore-tooling:
+```
+Microsoft.AspNetCore.Razor.LanguageServer.Common
+Microsoft.AspNetCore.Razor.LanguageServer
+Microsoft.AspNetCore.Razor.OmniSharpPlugin.StrongNamed
+Microsoft.AspNetCore.Razor.OmniSharpPlugin
+Microsoft.AspNetCore.Razor.VSCode.Extension
+Microsoft.AspNetCore.Razor.VSCode
+Microsoft.CodeAnalysis.Razor.Workspaces
+Microsoft.CodeAnalysis.Remote.Razor
+Microsoft.VisualStudio.Editor.Razor
+Microsoft.VisualStudio.LanguageServerClient.Razor
+Microsoft.VisualStudio.LanguageServices.Razor
+Microsoft.VisualStudio.LiveShare.Razor
+Microsoft.VisualStudio.Mac.LanguageServices.Razor
+Microsoft.VisualStudio.Mac.RazorAddin
+Microsoft.VisualStudio.RazorExtension
+RazorDeveloperTools
+rzls
+````
+THe following will be deleted:
+```
+RazorPageGenerator
+```
+
+Due to the separation of tooling projects, the new dependency flow will be:
+
+```
+runtime +--> aspnetcore +--> SDK
+        \               \
+         +-> extensions -+-> aspnetcore-tooling
+```
+### Known action items
+
+1. Migrate source with `git filter-branch`, preserving commit history
+2. Migration of pending PRs to aspnetcore after source migration.
+   1. Alternatively, we can freeze checkins during source migration especially since there isn't much source code that requires migration
+3. It seems like we do not need a pre-build step to build the Razor SDK for consumption elsewhere in the repo. It is possible to add a project dependency along with appropriate props/target imports
+4. Invert darc subscriptions so aspnetcore flows into aspnetcore-tooling
+
+## Phase two: Migration of tooling components
+
+After phase one is completed, we will migrate the tooling components remaining in aspnetcore-tooling piecemeal at a time as required. Our current understanding is that we will eventually be moving all of aspnetcore-tooling into aspnetcore but this can be scoped as appropriate given time/resource constraints. The overall goal here is to reduce build complexity in our repos and maintain developer productivity. As such, more discussion will be made with area owners to identify workflow impacts.
+
+## Major considerations
+
+### Pinning Roslyn dependencies
+
+Due to runtime and tooling divergence of Roslyn packages, we may need to pin multiple versions of Roslyn dependencies for the migrated packages.
+
+### Assets
+
+The migrated projects are C# only in phase one so we will use existing infrastructure in aspnetcore to handle signing and publishing of these packages. Phase two will involve additional asset types including vsix, zips, npm packages and mpacks. We have asset publishing mechanisms for most of these asset types but may need to add addtional asset publishing mechanisms and/or release management for some assets (mpacks for example). We will evaluate the additional requirements when phase two is discussed.
```

Comment:
These aren't really new requirements through, we've already solved them in the other repo (mpack and vscode as examples). Agreed that those don't need to come over right now.